react/finger-exercises/redux/02 - react-redux/src/app/screens/App/components/ShoppingCart/index.js

```diff
@@ -1,20 +1,17 @@
-import React, { PureComponent, Fragment } from 'react';
-import { arrayOf, func } from 'prop-types';
+import React, { Component, Fragment } from 'react';
+import { connect } from 'react-redux';
+import { arrayOf, func, bool } from 'prop-types';
 import { bookSelectedPropType } from '@constants/propTypes';
 import Button from '@components/Button';
 
+import actionsCreators from '../../../../../redux/shoppingCart/actions';
+
 import Item from './components/Item';
 import styles from './styles.scss';
 
-class ShoppingCart extends PureComponent {
-  state = {
-    open: false
-  };
-
+class ShoppingCart extends Component {
   toggleContent = () => {
-    this.setState(prevState => ({
-      open: !prevState.open
-    }));
+    this.props.toggleContent();
```

Comment:
This `toggleContent` prop can be passed directly to the `onClick`, then you can remove the `toggleContent` method and the component could be defined as a function.